hypex/dataset/datagenerator.py

```diff
@@ -0,0 +1,293 @@
+                                                                                   size=self.num_records)])
+        return info
+
+    def _generate_treatment_cols(self):
+        """
+        Generate columns with treatment values
+
+        Returns:
+            treatment:
+                Array with treatment
+        """
+        if self.is_treatment and self.num_targets == 1:
+            treatment = np.random.normal(0, self.sigma, (self.num_records, 1))
+        else:
+            treatment = np.random.normal(0, self.sigma, (self.num_records, self.num_targets))
+        treatment = np.digitize(treatment, bins=np.array([0, 0.5, 1])).astype(bool)
+        return treatment
+
+    def _generate_y_for_target(self, m_with_dummy):
+        """
+        Generate data for target field
+        Args:
+            m_with_dummy: Array with main features
+
+        Returns:
+            y0, y1: Arrays with values for target field
+        """
+        m_vector = m_with_dummy.dot(np.ones(self.num_features))
+        epsilon = np.random.multivariate_normal(mean=np.zeros(2),
+                                                cov=np.identity(2),
+                                                size=self.num_records)
+        y0 = m_vector + epsilon[:, 0]
+        y1 = self.sigma + m_with_dummy.dot(np.ones(self.num_features) * 2) + epsilon[:, 1]
+        return y0, y1
+
+    def _generate_outcome_and_full_data(self, m_with_dummy, treatment):
+        """
+        Generates outcome depending on treatment and create array with main numeric features
+        Args:
+            m_with_dummy: Array with main features
+            treatment: Array with treatment values
+
+        Returns:
+            data: Array with main numeric features
+        """
+        y0, y1 = self._generate_y_for_target(m_with_dummy)
+        if self.is_treatment and self.num_targets == 1:
+            outcome = (1 - treatment.flatten()) * y0 + treatment.flatten() * y1
+        else:
+            outcome = np.multiply((1 - treatment), y0[:, np.newaxis]) + np.multiply(treatment, y1[:, np.newaxis])
+            treatment = treatment[:, 0:1]
+
+        outcome = np.digitize(outcome, bins=np.array([np.mean(outcome) - 2])) if self.is_binary_target else outcome
+
+        if self.is_treatment:
+            return np.column_stack((m_with_dummy, treatment, outcome))
+        return np.column_stack((m_with_dummy, outcome))
+
+    def _set_df(self, data: np.ndarray, info: np.ndarray):
+        """
+        Generate names for columns and combine all data to DataFrame
+        Args:
+            data:
+                Array with main features
+            info:
+                Array with info features
+
+        Returns:
+            df:
+                DataFrame with generated data
+        """
+        gender = [["male", "female"][i] for i in np.random.choice(a=np.array([0, 1]), size=self.num_records)]
```

Comment:
`gender = np.random.choice(['male', 'female'], size=self.num_records)` it looks better